codec-http2/src/main/java/io/netty/handler/codec/http2/HpackHuffmanDecoder.java

```diff
@@ -4668,28 +4669,67 @@
             Http2Exception.newStatic(COMPRESSION_ERROR, "HPACK - Bad Encoding",
                     Http2Exception.ShutdownHint.HARD_SHUTDOWN), HpackHuffmanDecoder.class, "decode(..)");
 
+    private final HuffmanDecoderProcessor processor = new HuffmanDecoderProcessor();
+
+    HpackHuffmanDecoder() { }
+
     /**
      * Decompresses the given Huffman coded string literal.
      *
      * @param buf the string literal to be decoded
      * @return the output stream for the compressed data
      * @throws Http2Exception EOS Decoded
      */
-    public static AsciiString decode(ByteBuf buf, int length) throws Http2Exception {
-        if (length == 0) {
-            return AsciiString.EMPTY_STRING;
+    public AsciiString decode(ByteBuf buf, int length) throws Http2Exception {
+        return processor.decode(buf, length);
+    }
+
+    /**
+     * Using {@link ByteProcessor} to reduce bounds-checking and reference-count checking during byte-by-byte
+     * processing of the {@link ByteBuf}.
+     */
+    private static final class HuffmanDecoderProcessor implements ByteProcessor {
```

Comment:
Not sure how much hackery you want to go for, but you could make HpackHuffmanDecoder extend ByteProcessor.   The decode method would be the same, it would save two levels of call stack.  (one for `decode()`, one for the `access100()` methods for accessing the private variables.  Also it would save an object reference, and save one class in the jar file.  For a class that isn't going to be public, it probably won't be abused.

Reply:
+1 to this idea. This would also reduce object's footprint (if that matters).